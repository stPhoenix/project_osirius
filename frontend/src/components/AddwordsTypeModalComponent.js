import React from 'react';
import {Modal, ModalBody, Card, CardBody, CardTitle} from 'reactstrap';


export const AddwordsTypeModalComponent = (props) => {
    return (
        <Modal isOpen={props.modal} toggle={props.toggle_modal} className="p-0 align-self-center">
            <ModalBody className="p-0">
                <Card className="rounded-0">
                    <CardBody className="d-flex flex-column">
                        <CardTitle>Choose category</CardTitle>
                        
                            <form>
                                <select className="custom-form-control" id="learn_language" onChange={props.handleChange} name="category">
                                        {props.cats.map((category) => (<option key={category.name}>{category.name}</option>))}
                                </select>
                            </form>    
                        <button className="btn btn-link align-self-end" onClick={props.add_custom}>Add</button>
                    </CardBody>
                </Card>
            </ModalBody>
        </Modal>
    );
};