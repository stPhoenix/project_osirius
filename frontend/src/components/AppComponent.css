@font-face {
    font-family: 'exo_2bold';
    src: url('../assets/fonts/exo2-webfont.woff2') format('woff2'),
         url('../assets/fonts/exo2-webfont.woff') format('woff');
    font-weight: normal;
    font-style: normal;

}

.font-exo-2bold {
        font-family: 'exo_2bold';
}

.theme-primary{
    background-color: #009be5 !important;
}
.theme-light{
    background-color: #63ccff !important;
}
.theme-dark{
    background-color: #006db3 !important;
}
.bg-first{
	background-color: #FAFAFA !important;
}
.bg-second{
	background-color: #F5F5F5 !important;
}
html {
    background-color: #FAFAFA !important;
}
.vh-100{
	height: 100vh;
}

.full-screen{
    min-height: 100vh;
    max-height: 100%;
}
.custom-form-control {
  display: block;
  width: 100%;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  color: #495057;
  background-color: #f5f5f5;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  transition: border-color 0.15s ease-in-out;
}
.custom-form-control:focus {
  outline: 0;
  background-color: #ffffff;
}

.custom-btn {
  display: inline-block;
  font-weight: 400;
  text-align: center;
  color: white;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  font-size: 1rem;
  line-height: 1.5;
  transition: box-shadow 0.15s ease-in-out;
}

.custom-btn:focus, .custom-btn.focus {
  text-decoration: none;
  outline: 0;
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15)
}
.black {
    color: #000000;
}
.opacity-50{
    opacity: .5;
}

/* Customize the label (the container) */
.table-cell {
}

.m-checkbox-container {
  display: flex;
  position: relative;
  cursor: pointer;
  font-size: 22px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  
}

/* Hide the browser's default checkbox */
.m-checkbox-container input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
}

/* Create a custom checkbox */
.m-checkbox {
  position: absolute;
  height: 25px;
  width: 25px;
  background-color: #f5f5f5;
  border: 1px solid #ced4da;
}


/* When the checkbox is checked, add a blue background */
.m-checkbox-container input:checked ~ .m-checkbox {
    content: "";
    background-color: #009be5;
    border: none;
    background-image: url("data:image/svg+xml;charset=utf8,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 8 8'%3E%3Cpath fill='%23fff' d='M6.564.75l-3.59 3.612-1.538-1.55L0 4.26 2.974 7.25 8 2.193z'/%3E%3C/svg%3E");

}

/* Create the checkmark/indicator (hidden when not checked) */
.m-checkbox:after {
  content: "";
  position: absolute;
  display: none;
}

/* Show the checkmark when checked */
.m-checkbox-container input:checked ~ .m-checkbox:after {
  display: block;
  content: "";

}

.link-unstyled, .link-unstyled:visited, .link-unstyled:hover, .link-unstyled:active, .link-unstyled:focus, .link-unstyled:active:hover {
   text-decoration: none;
   color: white;
  }

.bg-overlay {
    background-color: rgba(0, 109, 180, 0.55);
}

.parallax-bg-1{
    background-image: url(../assets/bg2.jpg);
    background-repeat: no-repeat;
    background-size: cover;
    background-attachment: fixed;
    background-position: 0 0;
}

.parallax-bg-2{
    background-image: url(../assets/bg1.jpg);
    background-repeat: no-repeat;
    background-size: cover;
    background-attachment: fixed;
    background-position: 0 0;
}

.bg-banner{
    background-image: url(../assets/bg3.jpg);
    background-repeat: no-repeat;
    background-size: cover;
}

.btn-outline-white {
    border-style: solid;
    border-width: 3px;
    border-radius: 0;
    border-color: rgba(255, 255, 255, 1);
    background-color: transparent;
    color: rgba(255, 255, 255, 1);
}

.btn-outline-white:hover {
    background-color: rgba(255, 255, 255, 1);
    color: rgba(0, 109, 180, 1);
    transition: background-color 0.2s linear color 0.2s linear;
}