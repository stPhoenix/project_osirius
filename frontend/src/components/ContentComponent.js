import React from 'react';
import PropTypes from 'prop-types';

const componentProps = {
    children: PropTypes.oneOfType([
                                    PropTypes.array,
                                    PropTypes.object,
                                    PropTypes.string,
  ]),
};

export const ContentComponent = (props) => {
    return(
            <section className="row d-flex justify-content-center">
                <div className="col-12 col-md-11 col-xl-7 d-flex flex-column my-4 align-items-center">            
                    {props.children}
                </div>
            </section>
    );
};

ContentComponent.propTypes = componentProps;