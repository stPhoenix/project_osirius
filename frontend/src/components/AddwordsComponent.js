import React from 'react';
import {Link, Switch, Route} from 'react-router-dom';
import typing from '../assets/typing.png';
import presets from '../assets/presets.png';
import './AppComponent.css';
import AddwordsType from '../containers/AddwordsType';
import AddwordsPresets from '../containers/AddwordsPresets';
import {ContentComponent} from '../components';
import PropTypes from 'prop-types';

const componentProps = {
  location: PropTypes.object.isRequired,  
};

export const AddwordsComponent = ({location}) => {
	if (location.pathname !== "/addwords"){
		return (<Switch>
					<Route path="/addwords/type" component={AddwordsType} />
					<Route path="/addwords/presets" component={AddwordsPresets} />
				</Switch>);	
	};
	return(
    <ContentComponent>
            <section className="d-flex flex-column align-items-center justify-content-center">
		      <div className="d-flex flex-column align-items-center justify-content-center">
                <img src={typing} alt="typing" style={{height:'7rem', width:'auto'}} />
		        <h4 className="opacity-50 text-center px-2 mb-4 mt-0">Type word. System will search word in database or in the internet.</h4>
		        <Link className="custom-btn theme-primary px-3 py-1 mb-5 black" to="/addwords/type" > Add words</Link>
              </div>
              <div className="d-flex flex-column align-top align-items-center justify-content-center">
                <img src={presets} alt="presets" className="" style={{height:'6rem', width:'auto'}} />
		        <h4 className="opacity-50 text-center px-2 mb-4 mt-0">Choose word from preset in database.</h4>
		        <Link className="custom-btn theme-primary px-3 py-1 mb-3 black" to="/addwords/presets" > Add words</Link>
              </div>
            </section>
	</ContentComponent>
		);
};

AddwordsComponent.propTypes = componentProps;