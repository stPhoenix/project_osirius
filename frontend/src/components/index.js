import {AppComponent} from './AppComponent';
import {CustomAlertComponent} from './CustomAlertComponent';
import {LoginComponent} from './LoginComponent';
import {NavBarComponent} from './NavBarComponent';
import {NavItemComponent} from './NavItemComponent';
import {NewsComponent} from './NewsComponent';
import {SignUpComponent} from './SignUpComponent';

export {
    AppComponent,
    CustomAlertComponent,
    LoginComponent,
    NavBarComponent,
    NavItemComponent,
    NewsComponent,
    SignUpComponent
};