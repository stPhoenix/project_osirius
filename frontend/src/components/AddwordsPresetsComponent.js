import React from 'react';
import './AppComponent.css';
import {WordsTableComponent, ContentComponent} from '../components';


export const AddwordsPresetsComponent = (props) => {
    return (<ContentComponent>
                <form className="col-12 col-lg-8 px-0">
                    <label htmlFor="categories" >Choose category:</label>
                    <select className="custom-form-control" id="categories" onChange={props.handleChange} name="category">
                                        {props.cats.map((category) => (<option key={category.name}>{category.name}</option>))}
                    </select>
                </form>
                <div className="d-flex align-self-end mt-3 px-0" >
					<button className="btn btn-link " onClick={props.add_selected}>Add selected</button>
					<button className="btn btn-link pr-0" onClick={props.add_all}>Add all</button>
				</div>
                <WordsTableComponent words={props.words} handleCheck={props.handleCheck} />
          </ContentComponent>);  
};