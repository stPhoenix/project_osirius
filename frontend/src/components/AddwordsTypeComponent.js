import React from 'react';
import './AppComponent.css';
import {Card, CardBody, CardText} from 'reactstrap';
import {AddwordsTypeModalComponent, ContentComponent} from '../components';
import PropTypes from 'prop-types';

const componentProps = {
    global_word_search: PropTypes.bool,
    words: PropTypes.array,
    add_global: PropTypes.func,
    google_translate_search: PropTypes.bool,
    choose_category: PropTypes.func,
    search_word: PropTypes.string,
    handleChange: PropTypes.func,
    add_custom: PropTypes.func,
    cats: PropTypes.array,
    modal: PropTypes.bool,
    toggle_modal: PropTypes.func,
};

export const AddwordsTypeComponent = (props) => {
	
	let data = [];
	if (props.global_word_search === true){
		data = (props.words.map(word => {
			return (<div key={word.id+"words"} className="p-1">
                    <Card className="shadow-sm rounded-0">
                        <CardBody className="d-flex flex-column">
                            <CardText className="text-center">
                                <br />{word.name}
                                <br />{word.pronunciation}
                                <br />Category: [{word.category}]
                                <br />{word.translation}
                            </CardText>
                            <button name={word.id} className="btn btn-link align-self-end" onClick={props.add_global}>Add</button>
                        </CardBody>
                    </Card>
                </div>);
		}));
	} else if(props.google_translate_search === true){
		data = (props.words.map(word => {
			return (<div key={word.id+"words"} className="p-1">
                    <Card className="shadow-sm rounded-0">
                        <CardBody className="d-flex flex-column">
                            <CardText className="text-center">
                                <br />{word.name}
                                <br />{word.pronunciation}
                                <br />{word.translation}
                            </CardText>
                        <button className="btn btn-link align-self-end" onClick={(e) => {props.choose_category(word.name, word.pronunciation, word.translation)}}>Add</button>
                        </CardBody>
                    </Card>
                </div>  );
		}));
	}
	return(<ContentComponent>
            <h2>Add words by typying</h2>
				<form className="d-flex flex-column">
					<div className="form-group">
						<label htmlFor="search">Type word:</label>
						<input id="search" type="text" className="custom-form-control" value={props.search_word} name="search_word" onChange={props.handleChange}/>
					</div>
                    <button className="custom-btn theme-primary mb-2 align-self-center" onClick={props.search}>Search</button>
				</form>
                <div className="row flex-column flex-lg-row px-0">
                    {data}
                </div>
                <AddwordsTypeModalComponent add_custom={props.add_custom}
                                            handleChange={props.handleChange}
                                            cats={props.cats}
                                            modal={props.modal}
                                            toggle_modal={props.toggle_modal} />
		</ContentComponent>);
};

AddwordsTypeComponent.propTypes = componentProps;