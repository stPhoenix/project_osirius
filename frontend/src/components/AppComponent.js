import React from 'react';
import {Container, UncontrolledCarousel} from 'reactstrap';
import slide1 from '../assets/slide1.png';
import slide2 from '../assets/slide2.png';
import slide3 from '../assets/slide3.png';
import slide4 from '../assets/slide4.png';
import NavBar from '../containers/NavBar';
import Login from '../containers/Login';
import Alerts from '../containers/Alerts';
import {Switch, Route} from 'react-router';
import Logout from '../containers/Logout';
import SignUp from '../containers/SignUp';
import Settings from '../containers/Settings';
import ScrollToTop from '../assets/scroll-to-top.png';
import {AddwordsComponent, MywordsComponent, WordstrainerComponent, GetStartedComponent, PrivacyPolicyComponent, CookiePolicyComponent, DataCollectionComponent, FooterComponent, HomeComponent} from '../components';
import Learnwords from '../containers/Learnwords';
import Policy from '../containers/Policy';
import './AppComponent.css';
import {Link} from 'react-router-dom';


const items = [
    {
        src: slide1,
        altText: '',
        caption: '',
    },
    {
        src: slide2,
        altText: '',
        caption: '',
    },
    {
        src: slide3,
        altText: '',
        caption: '',
    },
    {
        src: slide4,
        altText: '',
        caption: '',
    },
];

export const AppComponent = () => {
    
    // When the user scrolls down 20px from the top of the document, show the button
        window.onscroll = () => {scrollFunction()};

        const scrollFunction = () => {
        if (document.body.scrollTop > 20 || document.documentElement.scrollTop > 20) {
            document.getElementById("scrollToTop").style.display = "flex";
        } else {
            document.getElementById("scrollToTop").style.display = "none";
        }
        }

        // When the user clicks on the button, scroll to the top of the document
        const topFunction = () => {
            const e = document.getElementsByTagName("body");
			e[0].scrollIntoView({block:"start", behavior: "smooth"});
        }  
        return (
      <Container fluid className="bg-first px-0">
        <div className="fixed-bottom m-3 justify-content-end mb-5" style={{display: "flex"}} id="scrollToTop">
            <img src={ScrollToTop} alt="Scroll To Top" className="theme-dark p-2" onClick={topFunction} />
        </div>
        <Policy />
        <header className="container-fluid sticky-top">
        <section className="row justify-content-center">
            <div className="col-12 col-lg-11 col-xl-7 theme-dark d-lg-flex shadow-sm">
                <NavBar />
            </div>
        </section>
    </header>
    <section className="container-fluid">
        <div className="row d-none d-lg-flex bg-banner mb-5">
            <div className="col-12 bg-overlay d-flex flex-column align-items-center font-exo-2bold text-white py-5">
                <h1 className="display-4 mt-5">HOW TO LEARN FOREIGN WORDS EASILY</h1>
                <h4 className="mb-5 mt-4">1. Sign Up  2. Add words  3. Learn words  4. Train words  5. Repeat step 2</h4>
            </div>
        </div>
    </section>
    <section className="container-fluid">
        <section className="row d-flex d-lg-none">
            <div className="col-12 px-0">
                 <UncontrolledCarousel items={items} />
            </div>
        </section>
        <section className="row parallax-bg-1 font-exo-2bold">
            <div className="col-12 bg-overlay d-flex justify-content-center px-0">
                <div className="col-12 col-lg-3 col-xl-2 d-flex flex-column my-5">
                    <Link className="btn btn-outline-white mb-2" to="/addwords">Add words</Link>
                    <Link className="btn btn-outline-white my-2" to="/learnwords">Learn words</Link>
                    <Link className="btn btn-outline-white my-2" to="/mywords">My words</Link>
                    <Link className="btn btn-outline-white mt-2" to="/wordstrainer">Train words</Link>
                </div>
            </div>
        </section>
    </section>
    <main className="container-fluid">
        <Alerts />
         <Switch>
             <Route exact path="/" component={HomeComponent} />
             <Route path="/login" component={Login} />
             <Route path="/logout" component={Logout} />
             <Route path="/signup" component={SignUp} />
             <Route path="/settings" component={Settings} />
             <Route path="/addwords" component={AddwordsComponent} />
             <Route path="/mywords" component={MywordsComponent} />
             <Route path="/learnwords" component={Learnwords} />
             <Route path="/wordstrainer" component={WordstrainerComponent} />
             <Route path="/getstarted" component={GetStartedComponent} />
             <Route path="/privacy_policy" component={PrivacyPolicyComponent} />
             <Route path="/cookie_policy" component={CookiePolicyComponent} />
             <Route path="/data_collection" component={DataCollectionComponent} />
            </Switch>
    </main>
    <FooterComponent />
    </Container>
    );
  };