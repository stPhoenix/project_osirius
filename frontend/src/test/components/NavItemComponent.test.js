
import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import {NavItemComponent} from '../../components';


describe('NavItemComponent', () => {
    it('should render correctly', () => {
        const componentProps = {
                                src: "#",
                                link: "",
                                text: "",
                                visible: true,
                                };
        const output = shallow(<NavItemComponent {...componentProps} />);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});