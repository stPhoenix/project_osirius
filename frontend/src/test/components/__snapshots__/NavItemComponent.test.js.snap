// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItemComponent should render correctly 1`] = `
<NavItem
  className="active"
  tag="li"
>
  <NavLink
    tag={[Function]}
    to=""
  />
</NavItem>
`;
